docs/usage/exposureclasses.md

```diff
@@ -0,0 +1,92 @@
+# ExposureClasses
+
+The Gardener API server provides a cluster-scoped `ExposureClass` resource.
+This resource is used to allow exposing the control plane of a Shoot cluster in various network enviroments like restricted corporate networks, dmzs etc.
+
+### Background
+
+The `ExposureClass` resource is based on the concept for the `RuntimeClass` resource in Kubernetes.
+
+A `RuntimeClass` abstracts the installation of a certain container runtime (e.g. gVisor, kata containers) on all nodes or a subset of the nodes in a Kubernetes cluster.
+See [here](https://kubernetes.io/docs/concepts/containers/runtime-class/).
+
+In contrast an `ExposureClass` abstracts the ability to expose a Shoot clusters control plane in certain network enviroments (e.g. corporate networks, dmz, internet) on all Seeds or a subset of the Seeds.
+
+Example: `RuntimeClass` and `ExposureClass`
+
+```yaml
+apiVersion: node.k8s.io/v1
+kind: RuntimeClass
+metadata:
+  name: gvisor
+handler: gvisorconfig
+# scheduling:
+#   nodeSelector:
+#     env: prod
+---
+kind: ExposureClass
+metadata:
+  name: internet
+handler: internetconfig
+# scheduling:
+#   seedSelector:
+#     matchLabels:
+#       network/env: internet
+```
+
+Similar like `RuntimeClasses` also `ExposureClasses` define a `.handler` field which reflects the name reference for the corresponding CRI configuration of the `RuntimeClass` and the control plane exposure configuration for the `ExposureClass`.
+
+The CRI handler for `RuntimeClasses` is usually installed by an admistrator (e.g. via a DemonSet which install the corresponding container runtime on the nodes).
+The control plane exposure configuration is managed by the Gardenlet as it need to take care that the control plane is exposed accordingly.
+See [here](#Gardenlet-Configuration-ExposureClass-handlers).
+
+
+The `RuntimeClass` support via its `.scheduling` section also the selection of a node subset (which have the respective controller runtime installed) for pod scheduling.
+The `ExposureClass` support also via the `.scheduling` section the selection of a subset of available Seed clusters whose Gardenlet is capable to apply the exposure configuration for the Shoot control plane accordingly.
+
+### Usage by a Shoot
+
+A `Shoot` can reference an `ExposureClass` via the `.spec.exposureClassName` field.
+
+**Attention:** When creating a Shoot cluster the Gardener scheduler will try to assign the Shoot cluster to a Seed cluster which will host its control plane.
+The scheduling behaviour can be influencd by the Shoots `seedSelectors` and/or `tolerations`.
+ExposureClasses can contain also scheduling instructions and if a Shoot is referencing an ExposureClass then the scheduling instructions of both will be merged into the Shoot.
+Those union of scheduling instructions might lead to a selection of a Seed which is not able to deal with the `handler` of the ExposureClass and the Shoot creation might end up in a creation error.
```

Comment:
```suggestion
Those unions of scheduling instructions might lead to a selection of a `Seed` which is not able to deal with the `handler` of the `ExposureClass` and the `Shoot` creation might end up in an error.
```

Reply:
Can you explain what needs to be done in this scenario? How to get rid of the error?

Reply:
I mean the user could modify the seedSelectors and/or tolerations of the Shoot to not interfere with the ones from the ExposureClass. If this is for whatever reason not possible then its might not possible to create the Shoot with this conditions and the error is correct.

Assuming the ExposureClass is only selecting AWS Seeds via `.scheduling.seedSelectors.providerTypes["aws"]`, but the user has its Shoot  a seed selector which is only allowing GCP Seeds via `.spec.seedSelectors.providerTypes["gcp"]`, then the union of provider types is `["aws", "gcp"]`. Assuming that only the AWS seeds have an ExposureClass handler for that ExposureClass and the Shoot get scheduled then to a GCP Seeds. it will lead to an error as the Seeds gardenlet does not have the right handler. If the user insists of scheduling to a `gcp` Seed then the scenario in combination with the ExposureClass is might not valid or need to contact an administrator.

Reply:
OK, maybe you can add this to the docs as well :)